packages/server/lib/cloud/protocol.ts

```diff
@@ -323,7 +323,17 @@ export class ProtocolManager implements ProtocolManagerShape {
           }
         }
 
-        const errorMessage = await res.json().catch(() => res.statusText)
+        const errorMessage = await res.json().catch(() => {
+          const url = new URL(uploadUrl)
```

Comment:
Can't imagine why this wouldn't be a valid url, but should we gaurd against if this ends up raising an error as invalid?

Reply:
While investigating this, I don't think we need to guard at this point, because this code is unreachable if `uploadUrl` is not a valid URL: `fetch` on L#305 will throw in this case, which gets caught on L#345.